File: backend/src/main/java/unpsjb/labprog/backend/model/Factura.java
package unpsjb.labprog.backend.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.JoinColumn;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.Date;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Factura {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    private Date fechaEmision;
    private Date fechaPago;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

}

File: frontend/cli/src/app/facturas/factura.ts
import { ArticulosPedido } from "../articulospedido/articulospedido";
import { Cliente } from "../clientes/cliente";

export interface Factura {
  id: number;

  fechaEmision: Date;

  fechaPago: Date;

  cliente: Cliente;

  total: number;

  articulosPedido: ArticulosPedido[];
}
